wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -0,0 +1,399 @@
+package com.squareup.wire.kotlin
+
+import com.squareup.kotlinpoet.BOOLEAN
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.DOUBLE
+import com.squareup.kotlinpoet.FLOAT
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.INT
+import com.squareup.kotlinpoet.KModifier.DATA
+import com.squareup.kotlinpoet.KModifier.FINAL
+import com.squareup.kotlinpoet.KModifier.OVERRIDE
+import com.squareup.kotlinpoet.KModifier.PRIVATE
+import com.squareup.kotlinpoet.LONG
+import com.squareup.kotlinpoet.NameAllocator
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asClassName
+import com.squareup.kotlinpoet.asTypeName
+import com.squareup.wire.EnumAdapter
+import com.squareup.wire.FieldEncoding
+import com.squareup.wire.Message
+import com.squareup.wire.ProtoAdapter
+import com.squareup.wire.ProtoReader
+import com.squareup.wire.ProtoWriter
+import com.squareup.wire.WireEnum
+import com.squareup.wire.schema.EnclosingType
+import com.squareup.wire.schema.EnumType
+import com.squareup.wire.schema.Field
+import com.squareup.wire.schema.MessageType
+import com.squareup.wire.schema.Options.ENUM_VALUE_OPTIONS
+import com.squareup.wire.schema.ProtoFile
+import com.squareup.wire.schema.ProtoMember
+import com.squareup.wire.schema.ProtoType
+import com.squareup.wire.schema.Schema
+import com.squareup.wire.schema.Type
+import okio.ByteString
+import java.util.*
+
+class KotlinGenerator private constructor(
+    val schema: Schema,
```

Comment:
Indentation on constructor params should be 2 spaces

Reply:
Cool, I will change this on the generated file too

Reply:
Is there a way to force intellij to use 2 spaces for constructor and function definitions ?

Reply:
https://github.com/square/java-code-styles